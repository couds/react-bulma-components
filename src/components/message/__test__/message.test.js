import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Message from '..';

describe('Message component', () => {
  it('Should Exist', () => {
    expect(Message).toMatchSnapshot();
  });
  it('Should have message classnames', () => {
    const component = renderer.create(
      <Message>
        <Message.Header>Lorem Ipsum</Message.Header>
        <Message.Body>Lorem Ipsum</Message.Body>
      </Message>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const component = renderer.create(
      <Message className="other-class">
        <Message.Header>Lorem Ipsum</Message.Header>
        <Message.Body>Lorem Ipsum</Message.Body>
      </Message>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render as an html section', () => {
    const component = renderer.create(
      <Message renderAs="section">This should be a section</Message>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const component = renderer.create(
      <Message renderAs="section" style={{ width: 200, zIndex: 1 }}>
        This should be a section with custom styles
      </Message>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should accept a react Element as renderAs prop', () => {
    // eslint-disable-next-line react/prop-types
    const Custom = props => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );
    Custom.propTypes = { children: PropTypes.node.isRequired };

    const component = renderer.create(
      <Message renderAs={Custom}>This should be a p element</Message>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
