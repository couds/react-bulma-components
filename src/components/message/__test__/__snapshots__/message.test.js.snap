// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Should Exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "Body": Object {
    "$$typeof": Symbol(react.forward_ref),
    "defaultProps": Object {
      "backgroundColor": undefined,
      "children": null,
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "hidden": undefined,
      "invisible": undefined,
      "italic": undefined,
      "marginless": undefined,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "div",
      "responsive": undefined,
      "shadowless": undefined,
      "textAlignment": undefined,
      "textColor": undefined,
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "unselectable": undefined,
    },
    "propTypes": Object {
      "backgroundColor": [Function],
      "children": [Function],
      "className": [Function],
      "clearfix": [Function],
      "clipped": [Function],
      "hidden": [Function],
      "invisible": [Function],
      "italic": [Function],
      "marginless": [Function],
      "overlay": [Function],
      "paddingless": [Function],
      "pull": [Function],
      "radiusless": [Function],
      "renderAs": [Function],
      "responsive": [Function],
      "shadowless": [Function],
      "textAlignment": [Function],
      "textColor": [Function],
      "textSize": [Function],
      "textTransform": [Function],
      "textWeight": [Function],
      "unselectable": [Function],
    },
    "render": [Function],
  },
  "Header": Object {
    "$$typeof": Symbol(react.forward_ref),
    "defaultProps": Object {
      "backgroundColor": undefined,
      "children": null,
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "hidden": undefined,
      "invisible": undefined,
      "italic": undefined,
      "marginless": undefined,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "div",
      "responsive": undefined,
      "shadowless": undefined,
      "textAlignment": undefined,
      "textColor": undefined,
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "unselectable": undefined,
    },
    "propTypes": Object {
      "backgroundColor": [Function],
      "children": [Function],
      "className": [Function],
      "clearfix": [Function],
      "clipped": [Function],
      "hidden": [Function],
      "invisible": [Function],
      "italic": [Function],
      "marginless": [Function],
      "overlay": [Function],
      "paddingless": [Function],
      "pull": [Function],
      "radiusless": [Function],
      "renderAs": [Function],
      "responsive": [Function],
      "shadowless": [Function],
      "textAlignment": [Function],
      "textColor": [Function],
      "textSize": [Function],
      "textTransform": [Function],
      "textWeight": [Function],
      "unselectable": [Function],
    },
    "render": [Function],
  },
  "defaultProps": Object {
    "backgroundColor": undefined,
    "children": null,
    "className": "",
    "clearfix": undefined,
    "clipped": undefined,
    "color": null,
    "hidden": undefined,
    "invisible": undefined,
    "italic": undefined,
    "marginless": undefined,
    "overlay": undefined,
    "paddingless": undefined,
    "pull": undefined,
    "radiusless": undefined,
    "renderAs": "article",
    "responsive": undefined,
    "shadowless": undefined,
    "size": null,
    "style": Object {},
    "textAlignment": undefined,
    "textColor": undefined,
    "textSize": undefined,
    "textTransform": undefined,
    "textWeight": undefined,
    "unselectable": undefined,
  },
  "propTypes": Object {
    "backgroundColor": [Function],
    "children": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "color": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "renderAs": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "size": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
  },
  "render": [Function],
}
`;

exports[`Message component Should accept a react Element as renderAs prop 1`] = `
<p
  className="message"
  style={Object {}}
>
  Custom
  This should be a p element
</p>
`;

exports[`Message component Should concat Bulma class with classes in props 1`] = `
<article
  className="message other-class"
  style={Object {}}
>
  <div
    className="message-header"
  >
    Lorem Ipsum
  </div>
  <div
    className="message-body"
  >
    Lorem Ipsum
  </div>
</article>
`;

exports[`Message component Should have custom inline styles 1`] = `
<section
  className="message"
  style={
    Object {
      "width": 200,
      "zIndex": 1,
    }
  }
>
  This should be a section with custom styles
</section>
`;

exports[`Message component Should have message classnames 1`] = `
<article
  className="message"
  style={Object {}}
>
  <div
    className="message-header"
  >
    Lorem Ipsum
  </div>
  <div
    className="message-body"
  >
    Lorem Ipsum
  </div>
</article>
`;

exports[`Message component Should render as an html section 1`] = `
<section
  className="message"
  style={Object {}}
>
  This should be a section
</section>
`;
